tests/test_llm_api_client_model_canonicalization.py

```
@@ -89,6 +89,56 @@ def _get_adapter(provider_id: str, model_config: ModelConfig) -> _FailingAdapter
     )
 
 
+@pytest.mark.asyncio
+async def test_api_client_surfaces_codex_chatgpt_model_availability_error(
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    class _FailingAdapter:
+        async def get_response(self, *args: Any, **kwargs: Any) -> str:
```

Ruff (0.15.11)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)